src/ResourceManager/Intune/Commands.Intune.Test/ScenarioTests/IntuneTestController.cs

```diff
@@ -0,0 +1,129 @@
+            }
+        }
+
+        /// <summary>
+        /// Parameterless constructor.
+        /// </summary>
+        public IntuneTestController()
+        {
+            IntuneTestController.InitializeEnvironment();
+            helper = new EnvironmentSetupHelper();
+        }
+
+        public void RunPsTest(params string[] scripts)
+        {
+            var callingClassType = TestUtilities.GetCallingClass(2);
+            var mockName = TestUtilities.GetCurrentMethodName(2);
+
+            RunPsTestWorkflow(
+                () => scripts,
+                // no custom cleanup 
+                null,
+                callingClassType,
+                mockName);
+        }
+
+
+        private void RunPsTestWorkflow(
+            Func<string[]> scriptBuilder,
+            Action cleanup,
+            string callingClassType,
+            string mockName)
+        {
+            using (MockContext context = MockContext.Start(callingClassType, mockName))
+            {
+                SetupManagementClients(context);
+                helper.SetupEnvironment(AzureModule.AzureResourceManager);
+
+                var callingClassName = callingClassType
+                                        .Split(new[] { "." }, StringSplitOptions.RemoveEmptyEntries)
+                                        .Last();
+                helper.SetupModules(AzureModule.AzureResourceManager,
+                    // "ScenarioTests\\Common.ps1", 
+                    "ScenarioTests\\" + callingClassName + ".ps1",
+                    helper.GetRMModulePath(@"AzureRM.Profile.psd1"),
+                    helper.GetRMModulePath(@"AzureRM.Resources.psd1"),
+                    helper.GetRMModulePath(@"AzureRM.Intune.psd1"));
+
+                try
+                {
+                    if (scriptBuilder != null)
+                    {
+                        var psScripts = scriptBuilder();
+
+                        if (psScripts != null)
+                        {
+                            helper.RunPowerShellTest(psScripts);
+                        }
+                    }
+                }
+                finally
+                {
+                    if (cleanup != null)
+                    {
+                        cleanup();
+                    }
+                }
+            }
+        }
+
+        private void SetupManagementClients(MockContext context)
+        {
+            IntuneManagementClient = GetIntuneManagementClient(context);
+
+            helper.SetupManagementClients(IntuneManagementClient);
+        }
+
+        private IntuneResourceManagementClient GetIntuneManagementClient(MockContext context)
+        {
+            return context.GetServiceClient<IntuneResourceManagementClient>(TestEnvironmentFactory.GetTestEnvironment());
+        }
+
+        internal static void InitializeEnvironment()
+        {
+            Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Playback");
```

Comment:
@rnikhilkumar Please remove this - this will; prevent you from recording tests without help.  This can wait until after the release, however.